rpcserver.go

```diff
@@ -269,6 +269,15 @@ func (s *rpcServer) Stop() error {
 	}
 
 	rpcLog.Info("Trader server stopping")
+
+	// TODO(guggero): Make sure this cannot lead to loss of channel if the
+	// trader shuts down after sending the sign message but before it can
+	// receive the finalize message.
+	if err := s.removePendingBatchArtifacts(); err != nil {
```

Comment:
So we don't always want to do this here as we could shut down after signing and before receiving the finalize message and end up deleting our channels (assuming the batch finalizes without further changes). Instead, this should go in `checkPendingBatch` where the other batch recovery logic lives after detecting a different batch was finalized (different batch transaction hash).